### PR TITLE
Improve AbstractTrainer Docs

### PR DESCRIPTION
**Pull request by Innixma**
*Issue #, if available:*

*Description of changes:*
- Improve AbstractTrainer Docs

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: core/src/autogluon/core/trainer/abstract_trainer.py [hunk 1]
```diff
@@ -90,20 +150,23 @@ class AbstractTrainer:
...
-        sample_weight=None,
-        weight_evaluation=False,
-        save_data=False,
-        random_state=0,
-        verbosity=2,
+        num_classes: int = None,
+        feature_metadata: FeatureMetadata = None,
+        eval_metric: Scorer = None,
+        quantile_levels: List[float] = None,
+        low_memory: bool = True,
```
Comment (Innixma) [inline core/src/autogluon/core/trainer/abstract_trainer.py:159]: Note: low_memory was previous default False, however in all cases AbstractTrainer was initialized, it was set to True. False is not actively supported, so I've updated the default to True here, should have no impact.

**Comment by yinweisu** [comment]
| Previous CI Run | Current CI Run |
| :---: | :---: |

**Review by prateekdesai04 — APPROVED** [review]
LGTM

**Comment by yinweisu** [comment]
| Previous CI Run | Current CI Run |
| :---: | :---: |

**Pull request merged**